backend/src/modules/doubles-tournament/core/application/event/CallMatchWhenMatchWasQueued.ts

```diff
@@ -0,0 +1,42 @@
+import { EventHandler } from '../../../../../shared/core/application/event/EventHandler';
+import { CommandPublisher } from '../../../../../shared/core/application/command/CommandBus';
+import { MatchesQueueRepository } from '../MatchesQueueRepository';
+import { TablesQueueRepository } from '../TablesQueueRepository';
+import { CallMatch } from '../command/CallMatch';
+import { MatchWasQueued } from '../../domain/event/MatchWasQueued';
+import { QueuedTable } from '../../domain/QueuedTable';
+
+export class CallMatchWhenMatchWasQueued implements EventHandler<MatchWasQueued> {
+  private readonly commandPublisher: CommandPublisher;
+  private readonly matchesQueueRepository: MatchesQueueRepository;
+  private readonly tablesQueueRepository: TablesQueueRepository;
+
+  constructor(
+    commandPublisher: CommandPublisher,
+    matchesQueueRepository: MatchesQueueRepository,
+    tablesQueueRepository: TablesQueueRepository,
+  ) {
+    this.commandPublisher = commandPublisher;
+    this.matchesQueueRepository = matchesQueueRepository;
+    this.tablesQueueRepository = tablesQueueRepository;
+  }
+
+  async handle(event: MatchWasQueued): Promise<void> {
+    const tournamentId = event.tournamentId;
+    const freeTable = await this.findFreeTables(tournamentId);
+    if (freeTable) {
```

Comment:
Chodzilo mi o to, że ten warunek jest taki "rozproszony". Bo nie ma nigdzie w kodzie, ze jesli jest freeTable i mecz nie wywołany. No i myslalem o tym, zeby zrobic jakas klase, w ktorej by sie to ogarnelo w stylu:

Ale to byłby refactor. Najwyzej potem. Wazne, ze testy dzialaja :) 
```
class CallingMatchProcess {

  constructor(private readonly tables: TablesQueue,
              private readonly matches: MatchesQueue
  ) {
  }

  handleTournamentTableWasReleased(event: TournamentTableWasReleased): { state: { tables: TablesQueue, match: MatchesQueue }, command: CallMatch | undefined } {
    //Tutaj zwracasz kolejki po operacji do zapisania i command, jesli warunki sa spelnione

  }

  handleMatchWasQueued(event: MatchWasQueued): { state: { tables: TablesQueue, match: MatchesQueue }, command: CallMatch | undefined } {
    //Tutaj zwracasz kolejki po operacji do zapisania i command, jesli warunki sa spelnione
    //Warunki sprawdzasz ifem ponizej


  }

  callMatchIfPossible(tables: TablesQueue, matches: MatchesQueue): { command: CallMatch | undefined } {
    if(//sprawdzasz czy tables i matches mają po wolnym stole i meczu - jak tak to tutaj zwracasz command)
  }
}
```